src/changelog/comment_changes.rs

```diff
@@ -280,7 +287,29 @@ impl Logic {
             },
             |r| {
                 self.repository = Some(r);
-                Ok(())
+
+                if let Some(tag) = &self.cli.tag {
+                    if let Some(repository) = &self.repository {
+                        if let Ok(target) =
+                            repository.resolve_reference_from_short_name(tag)
+                        {
+                            if target.is_tag() {
+                                self.stop_at_tag_oid = target.target();
+                                Ok(())
+                            } else {
+                                eprintln!("{tag} does not seem to be a tag.");
+                                Err(ExitCode::Usage)
+                            }
```

Comment:
The user might specify any valid Git reference, including branch names.  As this stop condition shall listen exclusively to tag names, this filter is required in order to prevent unexpected results.